Mathlib/CategoryTheory/Preadditive/AdditiveFunctor.lean

```diff
@@ -318,4 +320,45 @@ end Exact
 
 end Preadditive
 
+namespace Equivalence
+
+universe uC vC uC' vC' uD vD uD' vD'
+
+variable {C : Type uC} [Category.{vC} C] [Preadditive C]
+variable {C' : Type uC'} [Category.{vC'} C'] [Preadditive C']
+variable {D : Type uD} [Category.{vD} D] [Preadditive D]
+variable {D' : Type uD'} [Category.{vD'} D'] [Preadditive D']
+variable {f : C ⥤ D}  {g : C' ⥤ D'}
+variable {e : C ≌ C'} {e' : D ≌ D'} [e.functor.Additive] [e'.functor.Additive]
+
+/--
+Suppose we have categories `C, C', D, D'` such that the diagram of functors
+```
+C ===== f =====> D
+||e            ||e'
+||             ||
+C' ==== g ====> D'
+```
+commutes where `e` and `e'` are additive equivalence of categories.
+
+Then we have an isomorphism of endomorphism rings `End f ≃+* End g'` and
+-/
+noncomputable def endRingEquiv
+    (sq₀ : e.congrLeft.functor.obj f ≅ e'.congrRight.inverse.obj g) : End f ≃+* End g where
+  __ := endMonoidEquiv sq₀
+  map_add' α β := by
```

Comment:
[stacks#0DLP](https://stacks.math.columbia.edu/tag/0DLP) says that a functor between preadditive categories [with binary biproducts](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Limits/Shapes/Biproducts.html#CategoryTheory.Limits.HasBinaryBiproducts) that preserves either binary coproducts or binary products ([`Limits.PreservesLimitsOfShape (Discrete Limits.WalkingPair)`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Closed/Ideal.html#CategoryTheory.preservesBinaryProducts_of_exponentialIdeal)) (in particular if it's an equivalence) is automatically additive. Can you refactor the proof through that? That would also enable us to remove the additivity condition from the definition of Morita equivalence.